subprojects/platform-jvm/src/main/java/org/gradle/jvm/toolchain/internal/SharedJavaInstallationRegistry.java

```diff
@@ -76,6 +78,7 @@ private void finalizeValue() {
             .flatMap(Set::stream)
             .filter(this::installationExists)
             .map(InstallationLocation::getLocation)
+            .map(canonicalizer::canonicalize)
```

Comment:
Given what this method does, I wonder if it makes sense to pull the `native` project as a dependency just for that.

Did I miss something? We should be able to use Java 8 APIs in here.

Reply:
Yeah, I wasn't sure about that either; one thing that struck me is that we have a jdk7 fallback, not sure if it's worth trying to support toolchains on jdk7. And even then, that whole service sounds like something belonging to base services, not a native project. WDYT?